importlab/fs.py

```diff
@@ -118,17 +118,18 @@ def refer_to(self, path):
 
 
 class ExtensionRemappingFileSystem(RemappingFileSystem):
-    """File system that remaps .py file extensions."""
+    """File system that remaps file extensions."""
 
     def __init__(self, underlying, extension):
         super(ExtensionRemappingFileSystem, self).__init__(underlying)
         self.extension = extension
 
     def map_path(self, path):
-        p, ext = os.path.splitext(path)
-        if ext == '.py':
+        if self.underlying.isdir(path):
+            return path
+        else:
+            p, _ = os.path.splitext(path)
```

Comment:
if we are not hardcoding the from extension to .py, we should change self.extension to self.extension_map, and pass in {'py' : 'pyi'} when we use it in pytype. (this would help with the .py3 feature request too)

Reply:
I ended up reverting the change to this file because I realized that even if we're allowing things like scripts with no extension, it still makes sense for now to only map .py files.